src/main/java/com/faforever/client/map/generator/MapGeneratorService.java

```diff
@@ -91,14 +82,17 @@ public void postConstruct() {
 
 
   private void deleteGeneratedMaps() {
-    logger.info("Deleting leftover generated maps...");
+    log.info("Deleting leftover generated maps...");
 
     if (customMapsDirectory != null && customMapsDirectory.toFile().exists()) {
-      Arrays.stream(customMapsDirectory.toFile().listFiles())
-          .filter(File::isDirectory)
-          .filter(f -> GENERATED_MAP_PATTERN.matcher(f.getName()).matches())
-          .map(File::toPath)
-          .forEach(f -> noCatch(() -> FileUtils.deleteRecursively(f)));
+      try {
+        Files.list(customMapsDirectory)
```

Comment:
Files.list should be in the try-with-resources block otherwise the stream will not be closed

Reply:
No that is not case have a deeper look into what it does and you will see
```
    public static Stream<Path> list(Path dir) throws IOException {
        DirectoryStream<Path> ds = Files.newDirectoryStream(dir);
        try {
            final Iterator<Path> delegate = ds.iterator();

            // Re-wrap DirectoryIteratorException to UncheckedIOException
            Iterator<Path> iterator = new Iterator<>() {
                @Override
                public boolean hasNext() {
                    try {
                        return delegate.hasNext();
                    } catch (DirectoryIteratorException e) {
                        throw new UncheckedIOException(e.getCause());
                    }
                }
                @Override
                public Path next() {
                    try {
                        return delegate.next();
                    } catch (DirectoryIteratorException e) {
                        throw new UncheckedIOException(e.getCause());
                    }
                }
            };

            Spliterator<Path> spliterator =
                Spliterators.spliteratorUnknownSize(iterator, Spliterator.DISTINCT);
            return StreamSupport.stream(spliterator, false)
                                .onClose(asUncheckedRunnable(ds));
        } catch (Error|RuntimeException e) {
            try {
                ds.close();
            } catch (IOException ex) {
                try {
                    e.addSuppressed(ex);
                } catch (Throwable ignore) {}
            }
            throw e;
        }
    }
```

Reply:
Where is ds being closed? Only in the catch block.

Reply:
can u close streams did not know that

Reply:
Yes. Stream concepts are older than the Java 8 streams. A network connection is a stream, unix stdin or stdout are streams. These streams don't just close if you reach the end (some never end).

Stream extends BaseStream, BaseStream implements AutoCloseable. AutoCloseable defines the close method. AutoCloseable is the interface that is required to be used in the try-with-resources block.

So: collection streams are automatically cleaned up. File streams not necessarily.